### PR TITLE
Begin creating UAP backend classes

### PR DESCRIPTION
**Pull request by MCGallaspy**
## Summary

Building on #11, starts to flesh out the authentication (and authorization) backends.

## TODO

- [x] Have tests been written for the new code?
- [x] Has documentation been written/updated?
- [x] ~~New dependencies (if any) added to requirements file~~ (None added)
- [x] Add an entry to CHANGELOG.rst
- [x] Add yourself it AUTHORS.rst if you don't appear there

## Reviewer guidance

The main action is the the backends.py file, as well as test_backend.py. If #11 is merged, the diff will be considerably smaller.

## Issues addressed

None.

## Documentation

See https://docs.google.com/document/d/1BcOSxhoAOZGKjg9e7C6PpeQfHj9zkDqrKIk1WrbLsDU/edit

### CODE REVIEW

**Comment by MCGallaspy** [comment]
Rebased onto master after #11 was merged.

**Comment by MCGallaspy** [comment]
@benjaoming suggested creating new default device managers for the proxy models, let me address that before merging.

File: kolibri/auth/backends.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
+"""
+Implements a custom auth backend as described in the Django docs. Notably authentication could be provided by the
+default Django backend, but authorization (i.e. permissions checking) must be handled by our custom backends since we
+do not use the regular Django permissions. See handling authorization docs. Should then be listed in the
+AUTHENTICATION_BACKENDS. Note that authentication backends are checked in the order they're listed.
+"""
+from kolibri.auth.models import BaseUser, DeviceOwner, FacilityUser
+
+
+class BaseBackend(object):
```
Comment (benjaoming) [inline kolibri/auth/backends.py:10]: Currently, doesn't it do the same as the default backend?

File: kolibri/auth/migrations/0002_baseuser__is_device_owner.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (benjaoming) [inline kolibri/auth/migrations/0002_baseuser__is_device_owner.py]: Could probably reset migrations? :)

**Comment by benjaoming** [comment]
Tests!! :+1:

**Comment by MCGallaspy** [comment]
This is ready.

File: kolibri/auth/backends.py (hunk 1 above)
Comment (MCGallaspy) [inline kolibri/auth/backends.py:10]: More or less, but a little stripped down.

**Pull request merged**